patches/minecraft/net/minecraft/world/item/MobBucketItem.java.patch

```diff
@@ -1,43 +1,67 @@
 +   private final java.util.function.Supplier<? extends SoundEvent> emptySoundSupplier;
  
 +   @Deprecated
     public MobBucketItem(EntityType<?> p_151137_, Fluid p_151138_, SoundEvent p_151139_, Item.Properties p_151140_) {
-       super(p_151138_, p_151140_);
-       this.f_151134_ = p_151137_;
-       this.f_151135_ = p_151139_;
-+      this.emptySoundSupplier = null;
-+      this.fishTypeSupplier = () -> p_151137_;
+-      super(p_151138_, p_151140_);
+-      this.f_151134_ = p_151137_;
+-      this.f_151135_ = p_151139_;
++      this(() -> p_151137_, () -> p_151138_, () -> p_151139_, p_151140_);
 +   }
 +
 +   public MobBucketItem(java.util.function.Supplier<? extends EntityType<?>> entitySupplier, java.util.function.Supplier<? extends Fluid> fluidSupplier, java.util.function.Supplier<? extends SoundEvent> soundSupplier, Item.Properties properties) {
 +      super(fluidSupplier, properties);
-+      this.f_151134_ = null;
-+      this.f_151135_ = null;
 +      this.emptySoundSupplier = soundSupplier;
-+      this.fishTypeSupplier = entitySupplier;
++      this.entityTypeSupplier = entitySupplier;
     }
  
     public void m_142131_(@Nullable Player p_151146_, Level p_151147_, ItemStack p_151148_, BlockPos p_151149_) {
-@@ -80,6 +_,14 @@
+@@ -41,11 +_,11 @@
+    }
+ 
+    protected void m_7718_(@Nullable Player p_151151_, LevelAccessor p_151152_, BlockPos p_151153_) {
+-      p_151152_.m_5594_(p_151151_, p_151153_, this.f_151135_, SoundSource.NEUTRAL, 1.0F, 1.0F);
++      p_151152_.m_5594_(p_151151_, p_151153_, getEmptySound(), SoundSource.NEUTRAL, 1.0F, 1.0F);
+    }
+ 
+    private void m_151141_(ServerLevel p_151142_, ItemStack p_151143_, BlockPos p_151144_) {
+-      Entity entity = this.f_151134_.m_20592_(p_151142_, p_151143_, (Player)null, p_151144_, MobSpawnType.BUCKET, true, false);
++      Entity entity = getFishType().m_20592_(p_151142_, p_151143_, (Player)null, p_151144_, MobSpawnType.BUCKET, true, false);
+       if (entity instanceof Bucketable) {
+          Bucketable bucketable = (Bucketable)entity;
+          bucketable.m_142278_(p_151143_.m_41784_());
+@@ -55,7 +_,7 @@
+    }
+ 
+    public void m_7373_(ItemStack p_151155_, @Nullable Level p_151156_, List<Component> p_151157_, TooltipFlag p_151158_) {
+-      if (this.f_151134_ == EntityType.f_20489_) {
++      if (getFishType() == EntityType.f_20489_) {
+          CompoundTag compoundtag = p_151155_.m_41783_();
+          if (compoundtag != null && compoundtag.m_128425_("BucketVariantTag", 3)) {
+             int i = compoundtag.m_128451_("BucketVariantTag");
+@@ -80,6 +_,16 @@
              p_151157_.add(mutablecomponent);
           }
        }
 -
 +   }
 +
 +   // Forge Start
-+   protected EntityType<?> getFishType() {
-+      return fishTypeSupplier.get();
++   protected EntityType<?> getFishType()
++   {
++      return entityTypeSupplier.get();
 +   }
 +
-+   protected SoundEvent getEmptySound() {
++   protected SoundEvent getEmptySound()
++   {
```

Comment:
Because these are Minecraft classes, the code style to be followed is the same as the rest of the class (i.e. the formatting outputted by Fernflower): brackets on the same line.